src/main/java/org/opentripplanner/routing/alertpatch/EntitySelector.java

```diff
@@ -143,11 +166,16 @@ public int hashCode() {
   class StopAndRouteOrTripKey {
     private final FeedScopedId stop;
     private final FeedScopedId routeOrTrip;
+    private final ServiceDate serviceDate;
     private transient int hash = 0;
```

Comment:
Using 0 as NOT_SET is a bit weak since 0 is a valid hash, using -1 would be better.

Reply:
Fixed